state-sync/src/coordinator.rs

```diff
@@ -199,69 +198,112 @@ impl<T: ExecutorProxyTrait> StateSyncCoordinator<T> {
                 (network_id, event) = network_events.select_next_some() => {
```

Comment:
we should get rid of this network redirection nonsense and just be listening for network events directly....